src/aliceVision/calibration/checkerDetector.cpp

```diff
@@ -16,69 +22,74 @@ template <typename Type, int Size>
 using RowVector = Matrix<Type, 1, Size>;
 } // namespace Eigen
 
-namespace aliceVision{
-namespace calibration{
-
-
+namespace aliceVision {
+namespace calibration {
 
-bool CheckerDetector::process(const image::Image<image::RGBColor> & source)
+bool CheckerDetector::process(const image::Image<image::RGBColor>& source, bool useNestedGrids, bool debug)
 {
     image::Image<float> grayscale;
     image::ConvertPixelType(source, &grayscale);
 
+    const Vec2 center(grayscale.Width() / 2, grayscale.Height() / 2);
+
     const double scales[] = {1.0, 0.75, 0.5, 0.25};
 
-    std::vector<Vec2> allCorners;
+    std::vector<IntermediateCorner> allCorners;
     for (double scale : scales)
     {
+        ALICEVISION_LOG_INFO("[CheckerDetector] extracting corners at scale " << scale);
         std::vector<Vec2> corners;
         if (!processLevel(corners, grayscale, scale))
         {
+            ALICEVISION_LOG_DEBUG("[CheckerDetector] detection failed");
             return false;
         }
 
-        //Merge with previous level corners
+        ALICEVISION_LOG_DEBUG("[CheckerDetector] detected " << corners.size() << " corners");
+
+        // Merge with previous level corners
         for (Vec2 c : corners)
         {
-            double distmin = 5.0;
+            bool keep = true;
 
-            for (Vec2 oc : allCorners)
+            for (const IntermediateCorner & oc : allCorners)
             {
-                double dist = (oc - c).norm();
+                double dist = (oc.center - c).norm();
                 
-                if (dist < distmin)
+                if (dist < 5.0)
                 {
-                    distmin = dist;
+                    keep = false;
                     break;
                 }
             }
 
-            if (distmin < 5.0)
-            {
-                continue;
-            } 
+            if (!keep) continue;
 
-            allCorners.push_back(c);
+            IntermediateCorner sc;
+            sc.center = c;
+            sc.scale = scale;
+            allCorners.push_back(sc);
```

Comment:
cannot find the definition of `IntermediateCorner` but if it has the proper constructor you can just use `emplace_back`
```suggestion
            allCorners.emplace_back(c, scale);
```
be careful with the order of c and scale because it depends on the constructor definition, it could be the other way around

Reply:
Ok the definition was hidden by github, the order should be correct